Packs/Tenable_sc/Integrations/Tenable_sc/Tenable_sc.py

```diff
@@ -53,116 +24,1033 @@ def get_server_url():
+        """
+        Send the request for list_zones_command.
+        Returns:
+            Dict: The response.
+        """
+        return self.send_request(path='zone')
+
+    def get_scan_report(self, scan_results_id):
+        """
+        Send the request for get_scan_report_command and launch_scan_report_command.
+        Args:
+            scan_results_id (str): The ID of the scan_results to search for.
+        Returns:
+            Dict: The response.
+        """
+        path = 'scanResult/' + scan_results_id
 
-    session_cookie = cookies.create_cookie('TNS_SESSIONID', COOKIE)
-    SESSION.cookies.set_cookie(session_cookie)  # type: ignore
+        params = {
+            'fields': 'name,description,details,status,scannedIPs,progress,startTime,scanDuration,importStart,'
+                      'finishTime,completedChecks,owner,ownerGroup,repository,policy'
+        }
 
-    res = SESSION.request(method, url, data=json.dumps(body), params=params, headers=headers, verify=VERIFY_SSL)
+        return self.send_request(path, params=params)
 
-    if res.status_code == 403 and try_number <= MAX_REQUEST_RETRIES:
-        login()
-        headers['X-SecurityCenter'] = TOKEN  # The Token is being updated in the login
-        return send_request(path, method, body, params, headers, try_number + 1)
+    def create_query(self, scan_id, tool):
+        """
+        Send the request for get_vulnerabilities.
+        Args:
+            scan_id (str): The ID of the scan_results to create the query for.
+            tool (str): the tool to use.
+        Returns:
+            Dict: The response.
+        """
+        path = 'query'
 
-    elif res.status_code < 200 or res.status_code >= 300:
-        try:
-            error = res.json()
-        except Exception:
-            # type: ignore
-            return_error(
-                f'Error: Got status code {str(res.status_code)} with {url=} \
-                 with body {res.content} with headers {str(res.headers)}')   # type: ignore
+        body = {
+            'name': 'scan ' + scan_id + ' query',
+            'type': 'vuln',
+            'tool': tool,
+            'scanID': scan_id
+        }
 
-        return_error(f"Error: Got an error from TenableSC, code: {error['error_code']}, \
-                     details: {error['error_msg']}")  # type: ignore
-    return res.json()
+        return self.send_request(path, method='post', body=body)
+
+    def delete_query(self, query_id):
+        """
+        Send the request for get_vulnerabilities.
+        Args:
+            query_id (str): The ID of the query to delete.
+        Returns:
+            Dict: The response.
+        """
+        if not query_id:
+            raise DemistoException('query id returned None')
+        path = 'query/' + str(query_id)
+        self.send_request(path, method='delete')
+
+    def get_analysis(self, query=None, scan_results_id=None, args={}, calling_command="get_vulnerability_command"):
```

Comment:
instead of calling_command, lets make this a boolean that describes the affect it has. Or we should split into multiple functions and call them only where relevant

Reply:
can change to called_from_ get_vulnerability_command, is that good?